packages/browser/src/core/queue/event-queue.ts

```diff
@@ -265,6 +266,13 @@ export class EventQueue extends Emitter {
       }
     }
 
+    inspectorHost.trace({
+      stage: 'enriched',
+      id: ctx.id,
+      event: ctx.event as any,
```

Comment:
Does it worth just passing in the whole `ctx` object instead. Given the `ctx` will have both `id` and the `event` ... also if in the future we add more to the context, we don't have to change the trace signature

Reply:
`ctx` also has the `logs` and `metrics` for each plugin that's invoked which seems useful for inspectHost :smile:

Reply:
Good point, again! Will change that up.

Reply:
This will require extensive refactoring. We can postpone this until it becomes a necessity _(i.e when we do show extended metrics and logs in the UI)_